R/pkg/inst/tests/testthat/test_sparkSQL.R

```diff
@@ -2070,6 +2072,16 @@ test_that("dapply() on a DataFrame", {
   rownames(expected) <- NULL
   expect_identical(expected, result)
 
+  result <- dapplyCollect(
+              df,
+              function(x) {
+                y <- x[x$a > 1, ]
+                y <- cbind(y, y$a + 1L)
```

Comment:
No, the R function can't take addtional arguments, as R worker don't know how to do it.
If the additional data is small, it can be captured in the closure of the R function. Otherwise, broadcast variable can be used.
Does this answer your question?